tests/tests/test_00_multi_tenancy.py

```diff
@@ -14,81 +14,53 @@
 from .mendertesting import MenderTesting
-from ..helpers import Helpers
 from . import artifact_lock
 
-class TestMultiTenancyEnterprise(MenderTesting):
-    def mender_log_contains_aborted_string(self):
-        expected_string = "deployment aborted at the backend"
+from testutils.infra.device import MenderDevice
 
-        for _ in range(60*5):
-            with settings(hide('everything'), warn_only=True):
-                out = run("journalctl -u mender | grep \"%s\"" % expected_string)
-                if out.succeeded:
-                    return
-                else:
-                    time.sleep(2)
+class TestMultiTenancyEnterprise(MenderTesting):
 
-        pytest.fail("deployment never aborted.")
 
     def test_token_validity(self, enterprise_no_client):
         """ verify that only devices with valid tokens can bootstrap
             successfully to a multitenancy setup """
 
         wrong_token = "wrong-token"
 
-        def wait_until_bootstrap_attempt():
-            mender_clients = enterprise_no_client.get_mender_clients()
-            if not env.host_string:
-                return execute(wait_until_bootstrap_attempt,
-                               hosts=mender_clients)
-            Helpers.ssh_is_opened(mender_clients)
-
-            for i in range(1, 20):
-                    with settings(hide('everything'), warn_only=True):
-                        out = run('journalctl -u mender | grep "bootstrapped -> authorize-wait"')
-                        if out.succeeded:
-                            return True
-                        time.sleep(20/i)
-            return False
-
-        def set_correct_tenant_token(token):
-            if not env.host_string:
-                mender_clients = enterprise_no_client.get_mender_clients()
-                return execute(set_correct_tenant_token,
-                               token,
-                               hosts=mender_clients)
-
-            run("sed -i 's/%s/%s/g' /etc/mender/mender.conf" % (wrong_token, token))
-            run("systemctl restart mender")
-
         auth.reset_auth_token()
         auth.new_tenant("admin", "bob@bob.com", "hunter2hunter2")
         token = auth.current_tenant["tenant_token"]
 
         # create a new client with an incorrect token set
         enterprise_no_client.new_tenant_client("mender-client", wrong_token)
 
-        if wait_until_bootstrap_attempt():
-            for _ in range(5):
-                time.sleep(5)
-                auth_v2.get_devices(expected_devices=0)  # make sure device not seen
-        else:
-            pytest.fail("failed to bootstrap device")
+        mender_device = MenderDevice(enterprise_no_client.get_mender_clients()[0])
+
+        mender_device.ssh_is_opened()
+        # TODO check this change with Kristian
+        mender_device.run(
+            'journalctl -u mender | grep "authentication request rejected server error message: Unauthorized"',
+            wait=70,
```

Comment:
I *think* it should be ok!

Reply:
I *think* journalctl can do grepping also, right?

Reply:
Good eye @kacf, I actualy forgot about this TODO...

Let me elaborate and double check with you guys that this is okey. The previous code was grepping for "bootstrapped -> authorize-wait", which during my local run of this test didn't show up and I decided to change the text to grep for. That made me think that the old test logic was wrong, but but just reading the code it seemed to be okey. See the old logic [here](https://github.com/mendersoftware/integration/blob/beb63462cd1cd6e3772e709258486fd3d59697dc/tests/tests/test_00_multi_tenancy.py#L58-L64).

The test passes and I think it is safe to merge it, but it is still a mystery to me...

Reply:
(removing the TODO in the upcoming rework)